### PR TITLE
Fix Sphinx warnings in C API doc

### PR DESCRIPTION
**Pull request by vstinner**
----
:books: Documentation preview :books:: https://cpython-previews--107274.org.readthedocs.build/

### CODE REVIEW

**Comment by vstinner** [comment]
Hum. I already created PR #107164 but this one is supposed to be more complete. But then I fixed more and more warnings in a local branch, and quickly I went to too many directions. Now I'm confused and I'm not sure how to split this work.

So instead I created PR #107278 to first fix warnings like ``c:identifier reference target not found: va_list``.

File: Doc/c-api/bool.rst [hunk 1]
```diff
@@ -13,7 +13,7 @@ are available, however.
 
 .. c:function:: int PyBool_Check(PyObject *o)
 
-   Return true if *o* is of type :c:data:`PyBool_Type`.  This function always
+   Return true if *o* is of type :c:data:`!PyBool_Type`.  This function always
```
Comment (serhiy-storchaka) [inline Doc/c-api/bool.rst:16]: Would not be better to add the documentation for PyBool_Type?

File: Doc/c-api/apiabiversion.rst [hunk 2]
```diff
@@ -60,7 +60,7 @@ See :ref:`stable` for a discussion of API and ABI stability across versions.
 
    Use this for numeric comparisons, e.g. ``#if PY_VERSION_HEX >= ...``.
 
-   This version is also available via the symbol :data:`Py_Version`.
+   This version is also available via the symbol :c:var:`Py_Version`.
```
Comment (serhiy-storchaka) [inline Doc/c-api/apiabiversion.rst:63]: See #107091. It is discussed whether we should use `:c:var:` or `:c:data:`.

#107063 also fixes this line.

**Comment by vstinner** [comment]
I reworked this change into a serie of changes: I created issue #107298. I abandon this PR.

**Pull request closed without merging**